app/main/controllers/template/rtmedia-filters.php

```diff
@@ -627,3 +627,127 @@ function rtmedia_like_html_you_and_more_like_callback( $like_count, $user_like_i
+					INNER JOIN $term_taxonomy_table ON ( $term_taxonomy_table.term_id = $terms_table.term_id )
+					INNER JOIN $term_relationships_table ON ( $term_relationships_table.term_taxonomy_id = $term_taxonomy_table.term_taxonomy_id AND $term_relationships_table.object_id = $posts_table.ID )";
+			$att_id = $wpdb->get_results( $sql, ARRAY_N );
+			foreach ( $att_id as $key => $value ) {
+				$x .= $value[0] . ',';
+			}
+			if ( isset( $x )  && $x ) {
+				$x = rtrim( $x, ',' );
+			}
+
+			$where .= ' ( ';
+			$where .= " $table_name.media_title = '" . $_REQUEST['search'] . "' ";
+			if ( $author_id ) {
+				if ( $author_id->data->ID ) {
+					$where .= " OR $table_name.media_author = '" . $author_id->data->ID . "' ";
+				}
+			}
+			$where .= " OR post_table.post_content LIKE '%" . $_REQUEST['search'] . "%'";
+			$where .= " OR $table_name.media_type = '" . $_REQUEST['search'] . "' ";
+			if ( isset( $x )  && $x ) {
+				$where .= " OR $table_name.id IN ( $x )";
+			}
+			$where .= ' ) ';
+
+		}
+	}
+
+	return $where;
+}
+
+add_filter( 'rtmedia-model-where-query', 'rtmedia_search_fillter_where_query', 10, 3 );
+
+
+function rtmedia_search_fillter_join_query( $join, $table_name ) {
+
+	global $wpdb;
+	$posts_table = $wpdb->posts;
+	$terms_table = $wpdb->terms;
+	$term_relationships_table = $wpdb->term_relationships;
+	$term_taxonomy_table = $wpdb->term_taxonomy;
+
+	if ( isset( $_REQUEST['search'] ) ) {
+		// if ( false == strpos( $_SERVER['REQUEST_URI'], 'attribute' ) ) {
+			$join .= "INNER JOIN $posts_table as post_table ON ( post_table.ID = $table_name.media_id AND post_table.post_type = 'attachment')";
+			   // $join .= "INNER JOIN $posts_table ON $table_name.media_id = $posts_table.ID";
+		// }
+
+		$request_url = explode( '/', $_SERVER['REQUEST_URI'] );
+		if ( isset( $_REQUEST['search_by'] ) && 'attribute' == $_REQUEST['search_by'] && ! in_array( 'attribute', $request_url )  ) {
+			$join .= " 	INNER JOIN $posts_table ON ( $posts_table.ID = $table_name.media_id AND $posts_table.post_type = 'attachment' )
+	                    INNER JOIN $terms_table ON ( $terms_table.slug IN ('" . $_REQUEST['search'] . "') )
+	                    INNER JOIN $term_taxonomy_table ON ( $term_taxonomy_table.term_id = $terms_table.term_id )
+	                    INNER JOIN $term_relationships_table ON ( $term_relationships_table.term_taxonomy_id = $term_taxonomy_table.term_taxonomy_id AND $term_relationships_table.object_id = $posts_table.ID ) ";
+		}
+	}
+	return $join;
+}
+
+add_filter( 'rtmedia-model-join-query', 'rtmedia_search_fillter_join_query', 11, 2 );
+
+
+function rtmedia_model_query_columns( $columns ) {
+
+	if ( isset( $_REQUEST['search'] ) && $_REQUEST['search'] ) {
+
+		if ( isset( $_REQUEST['search_by'] ) ) {
+			if ( 'media_type' == $_REQUEST['search_by'] ) {
+				if ( isset( $columns['media_type']['value'] ) && is_array( $columns['media_type']['value'] ) ) {
+					$columns['media_type']['value'] = [ $_REQUEST['search'] ];
```

Comment:
Short array syntax is available since 5.4
This will cause fatal error in PHP version 5.3 and lesser